simple_salesforce/api.py

```diff
@@ -411,6 +411,12 @@ def __init__(self, object_name, session_id, sf_instance, sf_version='27.0', prox
         self.request = requests.Session()
         self.request.proxies = proxies
 
+        self.headers = headers if headers else {
```

Comment:
If you create an `SFType` on your own and pass custom headers you will also need to pass in `Content-Type` and `Authorization` since this change introduces an or instead of a header merge.